src/planning/src/planning/cone_coloring2.cpp

```diff
@@ -0,0 +1,140 @@
+#include "planning/cone_coloring2.hpp"
+
+void ConeColoring::remove_duplicates(std::vector<Cone>& cones) {
+  std::unordered_set<Cone, std::hash<Cone>, ConeAproxEqual> unique_cones;
+  std::vector<Cone> result;
+  for (const auto& cone : cones) {
+    if (unique_cones.insert(cone).second) {
+      result.push_back(cone);
+    }
+  }
+
+  cones = result;
+}
+
+Position ConeColoring::expected_initial_cone_position(const Pose& car_pose,
+                                                      const TrackSide& track_side) {
+  double distance_to_car = 2.0;
+  if (track_side == TrackSide::LEFT) {
+    return Position{car_pose.position.x - distance_to_car * sin(car_pose.orientation),
+                    car_pose.position.y + distance_to_car * cos(car_pose.orientation)};
+  } else if (track_side == TrackSide::RIGHT) {
+    return Position{car_pose.position.x + distance_to_car * sin(car_pose.orientation),
+                    car_pose.position.y - distance_to_car * cos(car_pose.orientation)};
+  } else {
+    RCLCPP_ERROR(rclcpp::get_logger("ConeColoring"), "Invalid track side");
+    return Position{0, 0};
+  }
+}
+
+Cone ConeColoring::find_initial_cone(
```

Comment:
this method is the same as before, right? what is gonna stop the error we got when the car is near one border and oriented kind of sideways? You had an idea before but I think it was different than this, no?

Reply:
Now the initial cone is marked as visited after this is called the first time. The second time this is called it ignores the cone that was first selected.